client/src/components/Verejne/index.js

```diff
@@ -26,13 +32,23 @@ const Verejne = ({
   autocompleteOptions,
   toggleModalOpen,
   openedAddressId,
+  entitySearchValue,
+  setEntitySearchValue,
+  findEntities,
 }) => (
   <div className="wrapper">
     <div className="verejne-side-panel">
       <EntitySearch />
-      <FormGroup>
-        <Input type="text" placeholder={FIND_ENTITY_TITLE} onClick={toggleModalOpen} />
-      </FormGroup>
+      <Form onSubmit={findEntities}>
+        <FormGroup>
+          <Input
+            type="text"
+            placeholder={FIND_ENTITY_TITLE}
+            value={entitySearchValue}
+            onChange={(e) => setEntitySearchValue(e.target.value)}
```

Comment:
Please, move this to `withHandlers`, because that arrow causes unnecessary re-renders of `Input` component.